sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/LeasedMeasurement.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.internal.state;
+
+import io.opentelemetry.api.common.Attributes;
+import javax.annotation.Nullable;
+
+/**
+ * A mutable {@link Measurement} implementation, that is leased to other objects temporarily.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ *
+ * <p>This class is not thread-safe.
+ */
+public class LeasedMeasurement implements Measurement {
```

Comment:
Since this isn't leased and returned, can we name it `MutableMeasurement` to match the other mutable classes?